axiom/nr_segment.c

```diff
@@ -1170,23 +1170,59 @@ void nr_segment_record_exception(nr_segment_t* segment,
   nr_segment_set_error(segment, error_message, error_class);
 }
 
-void nr_segment_set_error(nr_segment_t* segment,
-                          const char* error_message,
-                          const char* error_class) {
-  if ((NULL == segment) || (NULL == error_message && NULL == error_class)) {
-    return;
-  }
-
+static void nr_segment_set_error_helper(nr_segment_t* segment,
```

Comment:
I would recommend putting a comment on each of these static helper functions throughout this file to make it clear no NULL checking is done.  It is unlikely any new code would call these directly but I think it is a good safety guard to make sure a coder is aware of this.

Do you think we need unit tests for these helper functions throughout this file?

Reply:
I have removed all of the helper functions: [e177f40](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/e177f40a8b868410e5e1b728df7b5df8356ce1f4), [ed191c0](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/ed191c0b4ee2cff587aa3fcee1219c9133e586f7), [788c5a0](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/788c5a0e6330c7c788b2a6c5b028391319b7af17)